### PR TITLE
Rawphenotypes #11   code review

### PR DESCRIPTION
**Pull request by reynoldtan**
Solved Issue #11.

Please ignore commit item 10 in issue # 11 - Tripal Job Message. This is part of commit Non-code based issue in issue # 11.

To test:
Please clear your cache (Twice if possible) then click each pages in rawphenotypes module.

Rawdata page
Instructions page
Upload page
Backup page
Download page

Look at each page to see that it has no broken image, link etc. and if it has colour scheme plus a warning if module has no project nor data (if module was installed fresh). This is to test that new hook_preprocess that's been replaced with a more specific hook to form/page.

To test revised upload validators in file upload, simply upload a spreadsheet file into backup and upload, while doing so, see terminal to view shortened tripal jobs message.

More notes in each commit.

### CODE REVIEW

File: include/rawpheno.backup.form.inc [hunk 1]
```diff
@@ -165,7 +165,10 @@ function rawpheno_backup($form, &$form_state, $project_id = NULL, $command = 'de
         '#file_upload_max_size' => '10M',
         '#upload_location' => 'public://',
         '#upload_event' => 'manual',
-        '#upload_validators' => array('file_validate_extensions' => array('xlsx xls')),
+        '#upload_validators' => array(
+          'file_validate_extensions' => array('xlsx xls'),
+          'rawpheno_validate_spreadsheet_file' => array($form_state),
```
Comment (laceysanderson) [inline include/rawpheno.backup.form.inc]: This is an interesting approach I hadn't thought of --good idea! 
I think for file validators the first argument provided to the function is always $file and you only need to pass in arguments above if you need more then that. Thus you should be able to pass in nothing here: `'rawpheno_validate_spreadsheet_file' => array(),` and still use `function rawpheno_validate_spreadsheet_file($file) { }` below.

File: rawpheno.module [hunk 2]
```diff
@@ -252,114 +252,151 @@ function rawpheno_permission() {
...
+  // Version
+  $variables['d3ver'] = (isset($ver[0])) ? trim($ver[0]) : 0;
+  // Release
+  $variables['d3rel'] = (isset($ver[1])) ? trim($ver[1]) : 0;
+
+  // Does D3 exists
+  $variables['d3_in'] = libraries_load('d3js');
+}
+
+// Instruction page.
```
Comment (laceysanderson) [inline rawpheno.module]: Each function still needs a doc block. For example,
```
/**
 * Implements hook_preprocess_rawpheno_instructions().
 * This function prepares variables for the Raw Phenotypes Instruction page.
 */
```

File: include/rawpheno.backup.form.inc (hunk 1 above)
Comment (reynoldtan) [inline include/rawpheno.backup.form.inc]: Lacey, I need to do more test on this. It appears that both upload and backup freeze when spreadsheet has no measurements tab.

Comment (reynoldtan) [inline include/rawpheno.backup.form.inc]: This issue occurs when spreadsheet has an error. I appears to freeze in backup, while it won't let user to re-upload in upload page. After examining the code, I think custom validators are executed before form submit, thus errors show but the upload Drag and drop does not, whereas hook_file_validate() submits the form (reloads the dnd form) plus does the validation check, hence when error is present we see both error and the form.

Restored to using hook_file_validate().

**Pull request merged**